test/function_test.ml

```diff
@@ -0,0 +1,88 @@
+open! Core
+open! Ctypes
+
+(* Tests for Scalar_function module *)
+
+let%expect_test "scalar_function_basic" =
+  Duckdb.Database.with_path ":memory:" ~f:(fun db ->
+    Duckdb.Connection.with_connection db ~f:(fun conn ->
+      (* Set DuckDB to single-threaded mode to avoid thread safety issues *)
+      Duckdb.Query.run_exn' conn "SET threads TO 1";
+      (* Create a simple scalar function that doubles an integer *)
+      let double_function =
+        Duckdb.Scalar_function.create
+          "double_int"
+          (Integer :: Returning Integer)
+          ~f:(fun x -> Int32.(x * 2l))
+      in
+      (* Register the function *)
+      Duckdb.Scalar_function.register_exn double_function conn;
+      (* Test the function *)
+      Duckdb.Query.run_exn conn "SELECT double_int(42)" ~f:(fun res ->
+        Duckdb.Result_.to_string_hum res ~bars:`Unicode |> print_endline);
```

Comment:
It feels like there's a lot of duplication of `Duckdb.Result_.to_string_hum ...` call. Let's extract it out into a helper function.

Reply:
I don't think this has been addressed.

Reply:
I don't think this has been addressed.

Reply:
I don't think any calls to `(fun res -> Duckdb.Result_.to_string_hum res ~bars:`Unicode |> print_endline)` should remain now that we have helper functions.